### DISCUSSION TITLE
Set box_type:"cold", and make use of elasticsearch ILM

### DISCUSSION THREAD

**Discussion by astroc0** [Security-Onion-Solutions/securityonion#6553] [Unsupported Versions] [answered]
Our goal is to add a separated search node that contain big pool of storage disks to move shards of indices getting older than 7 days to store it there permanently.
Previously we could do that easily by adding a cold storage node for elasticsearh and setup ILM policy to move indices older than 7 days to it while marking it read-only to save some space.
 
**How we can do that in SO 2???**

**Comment by TOoSmOotH** [comment, marked as answer]
https://docs.securityonion.net/en/2.3/elasticsearch.html#customization

Nothing is preventing you from using ILM. You can change any setting in the minion pillar for elasticsearch.

**Reply by astroc0** [reply]
Oh my bad.
One more question, how can we set box_type:"cold" in minion pillar file? how it will like like?
Is the following code correct to add in minion pillar?
```
elasticsearch:
  config:
    node.attr.box_type: cold
```
or it have be like this..?
```
elasticsearch:
  config:
    node:
      attr:
        box_type: cold
```

**Reply by astroc0** [reply]
the first code works

**Comment by astroc0** [comment]
After linking the index template to the ILM policy created, it gets reset every salt highstate, and the assignment will be gone!
**What is the proper way to link so-zeek-* "or so-*" index pattern with specific ILM policy???**

**Reply by astroc0** [reply]
I know I could be naive pinging you @TOoSmOotH, but hope you look at it asap. thank you!

**Reply by astroc0** [reply]
Regarding this, we would love to see some intention soon.

Last thing we tried modifying the so-common and so-zeek index templates after copying it to local from default, added the name for the ilm policy we created: `index.lifecycle.name: "zeek_ilm" `, then we can verify its applied in elasticsearch.yml but we don't see in kibana that the index template is linked with the ilm policy.
Consider clustering and data allocation is one of main features distributed deployment is used..

Also maybe you can add the following steps regarding cold/hot node assignment for documentation:
```
elasticsearch:
  config:
    node.attr.box_type: cold
```

**Reply by astroc0** [reply]
bump :)

**Reply by astroc0** [reply]
bump!

**Reply by astroc0** [reply]
UPDATE: once we rebooted the manager the policy we made shows finally that it has some #of linked indices!
_**Wonder why so many salt updates didn't resolve it until rebooted.**_

**Comment by dougburks** [comment]
@astroc0 We are quite busy dealing with the recent log4j vulnerability and that is our top priority right now. There is no need to ping anybody to get their attention. Please be patient and review the community guidelines:
https://github.com/Security-Onion-Solutions/securityonion/discussions/1720

Thank you.

**Reply by astroc0** [reply]
Thanks to all of your team for the great work you do!

Our team is so impressed with SO, the free and open system that strongly competes with big names of enterprise siem/soc/noc solutions out there!
